### PR TITLE
Psalm plugins revamp + PHPUnit, Symfony

### PR DESCRIPTION
**Pull request by dkarlovi**
1. Psalm moved from PHAR to be installed via Composer (AFAIK, required for the plugins to be found)
2. moved the existing Doctrine plugin to the same namespace, `psalm`
3. added Symfony plugin, #235
4. added PHPUnit plugin, #99

### CODE REVIEW

**Comment by dkarlovi** [comment]
The issue here is described in https://github.com/vimeo/psalm/issues/3640

**Comment by jakzal** [comment]
Thank you for working on this.

Integrated tests are failing:

```
-e ✘psalm-plugin-doctrine
777
Makefile:32: recipe for target 'test-integration' failed
```

**Comment by dkarlovi** [comment]
@jakzal I know, the `psalm-plugin` tool (which I'd like to use to confirm Psalm can actually find the plugins) currently doesn't work without a config file, it's in the linked issue.

**Comment by dkarlovi** [comment]
Fixed the underlying issue in https://github.com/vimeo/psalm/pull/4639, but in the meantime Psalm 4 came out and plugins haven't caught up yet:
https://github.com/weirdan/doctrine-psalm-plugin/pull/78

**Comment by jakzal** [comment]
@dkarlovi could this issue be related? https://github.com/jakzal/phpqa/issues/263

**Comment by dkarlovi** [comment]
@jakzal in what way, since this isn't yet merged? :thinking: Or am I misunderstanding you?

**Comment by jakzal** [comment]
I was more asking if recent changes in psalm could cause that other issue?

**Comment by dkarlovi** [comment]
Not that I know of. Seems like something new, maybe an issue needs to be raised on Psalm repo directly.

**Comment by dkarlovi** [comment]
Seems Psalm Doctrine plugin now allows Psalm 4, this should be revisited soonish. There's also some indication the plugin might become officially supported by Psalm org and/or fully deprecated because of additions to Doctrine itself, see https://github.com/weirdan/doctrine-psalm-plugin/issues/84

**Comment by jakzal** [comment]
@dkarlovi can you rebase when you have few minutes?

**Comment by dkarlovi** [comment]
Sure, left as TODO, will do it tomorrow. :+1:

File: resources/psalm.json [hunk 1]
```diff
@@ -5,29 +5,58 @@
...
             "summary": "Stubs to let Psalm understand Doctrine better",
             "website": "https://github.com/weirdan/doctrine-psalm-plugin",
             "command": {
                 "composer-bin-plugin": {
                     "package": "weirdan/doctrine-psalm-plugin",
-                    "namespace": "doctrine-psalm-plugin"
+                    "namespace": "psalm"
+                }
+            },
+            "test": "rm -f composer.json; psalm-plugin show | grep weirdan/doctrine-psalm-plugin",
```
Comment (dkarlovi) [inline resources/psalm.json:33]: `psalm-plugin` works if
1. there's a `composer.json` and `vendor/`
2. both are missing
In our case, we have just `composer.json` so it's saying there should be an autoloader here too, even there is Psalm's one already.

Not sure what the better choice here would be, maybe changing to the Psalm namespace root and running it there?

**Comment by dkarlovi** [comment]
@jakzal please check now, not happy with the workaround mentioned, maybe you have a better idea?

File: resources/psalm.json (hunk 1 above)
Comment (jakzal) [reply resources/psalm.json:33]: > Not sure what the better choice here would be, maybe changing to the Psalm namespace root and running it there?

Either this, or change to a tmp (or root) directory.

Comment (dkarlovi) [reply resources/psalm.json]: @jakzal done.

**Comment by jakzal** [comment]
@dkarlovi since you renamed some tools, can you also regenerate the README with `php bin/devkit.php update:readme` please?

**Comment by jakzal** [comment]
@dkarlovi how risky is this change? :) 

We have a relatively big change coming in soon (#369), and I'm thinking about how to approach merges and releases. #369 should definitely go in on its own release, and then sit in phpqa's master for a little while before it's tagged in phpqa. I'm wondering if this change should follow a similar process.

**Comment by dkarlovi** [comment]
@jakzal IMO it's not very risky, the major difference is you now don't have two separate Psalm's (the PHAR which is not capable of using plugins and the one embedded into the `doctrine-psalm-plugin` namespace which is, but can only see the single plugin).

It will only affect people who currently use the Doctrine plugin in that they'll need to use `psalm` without the full path, so the migration path is very clear. Everyone else is unaffected.

**Comment by dkarlovi** [comment]
@jakzal what do you see as the way forward here?

**Comment by jakzal** [comment]
Thank you @dkarlovi 🍺

**Comment by dkarlovi** [comment]
BTW @jakzal this works, I've tried `jakzal/phpqa:alpine` just now, it loads the plugins correctly on my test project.

**Pull request merged**